megamek/src/megamek/utilities/QuickGameRunner.java

```diff
@@ -0,0 +1,383 @@
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+
+package megamek.utilities;
+
+import io.sentry.Sentry;
+import megamek.client.AbstractClient;
+import megamek.client.Client;
+import megamek.client.CloseClientListener;
+import megamek.client.HeadlessClient;
+import megamek.client.bot.princess.Princess;
+import megamek.client.ui.swing.*;
+import megamek.client.ui.swing.util.MegaMekController;
+import megamek.common.Game;
+import megamek.common.MMRandom;
+import megamek.common.MekSummaryCache;
+import megamek.common.Player;
+import megamek.common.enums.GamePhase;
+import megamek.common.event.GameListenerAdapter;
+import megamek.common.event.GamePhaseChangeEvent;
+import megamek.common.net.marshalling.SanityInputFilter;
+import megamek.common.options.IBasicOption;
+import megamek.common.options.OptionsConstants;
+import megamek.common.preference.PreferenceManager;
+import megamek.logging.MMLogger;
+import megamek.server.Server;
+import megamek.server.totalwarfare.TWGameManager;
+
+import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInputFilter;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static megamek.MMConstants.*;
+
+/**
+ * QuickGameRunner
+ * This class is used to run a game without minimal preparation
+ * all it needs is a game file to load and it will populate the bots in the game
+ * and run the game the predefined number of turns, or until the game ends or until
+ * it times out.
+ */
+public class QuickGameRunner {
+    private static final MMLogger logger = MMLogger.create(QuickGameRunner.class);
+
+    static {
+        MekSummaryCache.getInstance();
+        ObjectInputFilter.Config.setSerialFilter(new SanityInputFilter());
+    }
+
+    private Server server;
+    private Client client;
+    private final Random random = new Random();
+    private final GUIType guiType;
+
+    public enum GUIType {
+        DEFAULT,
+        COMMANDER,
+        NONE,
+    }
+
+    /**
+     * QuickGameRunner, initializes the game and runs it
+     * @param gameFile The game file to load
+     * @param GUIType The GUI type to use, Default is ClientGUI, Commander is CommanderGUI, None does not load a GUI at all.
+     * @param roundsLimit  The number of rounds to run the game
+     * @param timeoutMinutes  The maximum number of minutes to wait for the game to complete
+     */
+    public QuickGameRunner(File gameFile, GUIType GUIType, int roundsLimit, int timeoutMinutes) {
+        this.guiType = GUIType;
+        try {
+            initialize();
+            if (gameFile != null) {
+                run(gameFile, roundsLimit, timeoutMinutes);
+            }
+        } catch (Exception e) {
+            Sentry.captureException(e);
```

Comment:
This being a utility method, do you really want to have it send the errors to Sentry vs just deal with them locally when you run them? 

Or is the intention for when others run this for those same said errors to be reported?

Reply:
I added sentry by mistake here, but looking at the code inside error, there is no way to log error without calling sentry, I would expect any error to be reported, independent of where it was generated, however I could see it being too noisy if someone is using this in an automated manner.